File: Back/Forum/src/main/java/com/example/forum/Entity/Comment.java
package com.example.forum.Entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table
@Builder
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long idComment ;
    private int likesComment;
    private int dislikesComment;
    private String textComment;
    @Temporal(TemporalType.DATE)
    private Date dateCreationComment;
    private int userId;


    @JsonIgnore
    @ManyToOne
    Post post;
}

File: Front/src/app/models/comment.ts
import {Post} from './post';
import { User } from './user';

export class Comment{
  idComment!: number;
  likesComment!: number;
  mostPertinentComment!: boolean;
  textComment!: string;
  dateCreationComment!: string ;
  post!: Post;
  userId:number
  fullUser:User
}